2019/05/5.rb

```diff
@@ -0,0 +1,169 @@
+POSITIONMODE = "0"
+IMMEDIATEMODE = "1"
+
+def get_first_parameter_value(command, intcode, index)
+  first_input_index = Integer(intcode[index+1])
+  first_param_index_mode = command[-3]
+
+  if first_param_index_mode == IMMEDIATEMODE
+    first_input_value = first_input_index
+  else
+	first_input_value = Integer(intcode[first_input_index])
+  end
+  
+  return first_input_value
+end
+
+def get_second_parameter_value(command, intcode, index)
+  second_input_index = Integer(intcode[index+2])
+  second_param_index_mode = command[-4]
+  
+  if second_param_index_mode == IMMEDIATEMODE
+    second_input_value = second_input_index
+  else
+    second_input_value = Integer(intcode[second_input_index])
+  end
+  
+  return second_input_value
+end
+
+def get_third_parameter_value(command, intcode, index)
+  output_index = Integer(intcode[index+3])
```

Comment:
Not a problem here, but just a knowledge drop: Be a bit wary with parsing integers like this. Ruby might derive the number base on what it thinks is right

```ruby
Integer('0xAF')
 => 175
```

I mention that because of the puzzle domain of op codes